### PR TITLE
Only send an empty watch update for the initial get.

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

File: cpp/util/etcd.cc [hunk 1]
```diff
@@ -514,6 +515,7 @@ struct EtcdClient::WatchState {
 
   int64_t highest_index_seen_;
   map<string, int64_t> known_keys_;
+  bool first_update_;
```
Comment (pphaneuf) [inline cpp/util/etcd.cc]: Could figure that out locally at line 551, if ```highest_index_seen_``` is still -1?

File: cpp/util/etcd.cc [hunk 2]
```diff
@@ -581,8 +588,11 @@ void EtcdClient::WatchInitialGetAllDone(WatchState* state, util::Status status,
 
   state->known_keys_.swap(new_known_keys);
 
-  state->task_->executor()->Add(
-      bind(&EtcdClient::SendWatchUpdates, this, state, move(updates)));
+  if (!updates.empty() || state->first_update_) {
+    state->first_update_ = false;
+    state->task_->executor()->Add(
+        bind(&EtcdClient::SendWatchUpdates, this, state, move(updates)));
```
Comment (pphaneuf) [inline cpp/util/etcd.cc]: Can't do this! ```SendWatchUpdates``` sends the update... *and* starts the next request!

I was going to point out that there's two spots calling it, as well, so if you're going to do it this way, should do it centrally, in ```SendWatchUpdates```, but as it turns out, it's the only way to go.

File: cpp/util/etcd.cc (hunk 1 above)
Comment (AlCutter) [inline cpp/util/etcd.cc]: Nice, thanks.

File: cpp/util/etcd.cc (hunk 2 above)
Comment (AlCutter) [inline cpp/util/etcd.cc]: Ah - yes, thank you!
Cool, updated.

File: cpp/log/cluster_state_controller-inl.h [hunk 3]
```diff
@@ -446,11 +449,14 @@ void ClusterStateController<Logged>::DetermineElectionParticipation(
...
 void ClusterStateController<Logged>::ClusterServingSTHUpdater() {
   while (true) {
+    LOG(INFO) << "ClusterServingSTHUpdater going again.";
     std::unique_lock<std::mutex> lock(mutex_);
     update_required_cv_.wait(lock, [this]() {
       return update_required_ || exiting_;
     });
+    LOG(INFO) << "ClusterServingSTHUpdater got ping.";
     if (exiting_) {
+      LOG(INFO) << "ClusterServingSTHUpdater thread returning.";
```
Comment (pphaneuf) [inline cpp/log/cluster_state_controller-inl.h]: Not sure if those should be ```VLOG``` or not, but this is fine for now, we can change it later.

**Comment by pphaneuf** [comment]
LGTM, modulo adding detail to that log message (and Travis).

File: cpp/log/cluster_state_controller-inl.h (hunk 3 above)
Comment (AlCutter) [inline cpp/log/cluster_state_controller-inl.h]: They kinda slipped in there, but they might be useful again the in future so I've mostly cranked them down to `VLOG(1)`, a couple remain `LOG(INFO)` though.

**Pull request merged**